### ISSUE TITLE
Mixing static + wildcard segment causes hung request

### ISSUE THREAD

**Issue by ImVexed** [fasthttp/router#37]
```go
package main

import (
	"fmt"
	"log"

	"github.com/fasthttp/router"
	"github.com/valyala/fasthttp"
)

func Index(ctx *fasthttp.RequestCtx) {
	ctx.WriteString("Welcome!")
}

func Hello(ctx *fasthttp.RequestCtx) {
	fmt.Fprintf(ctx, "Hello, %s!\n", ctx.UserValue("name"))
}

func main() {
	r := router.New()
	r.GET("/hello/test", Index)
	r.GET("/hello/{name}", Hello)

	fmt.Println("starting")
	log.Fatal(fasthttp.ListenAndServe(":8080", r.Handler))
}
```

GET: /hello/test returns "Welcome!"
GET: /hello/test1 hangs indefinitely
GET: /hello/tes returns "Hello, tes!"

**Comment by savsgio** [comment]
Hi @ImVexed,

Thanks for your issue. I will fix it as soon as possible.

**Comment by savsgio** [comment]
Fixed!! 🎉 

The bug fix it's available in the new release [v1.2.4](https://github.com/fasthttp/router/releases/tag/v1.2.4).
